### ISSUE TITLE
GC on ShortShake3D

### ISSUE THREAD

**Issue by StinkySteak** [gasgiant/Camera-Shake#3]
![image](https://github.com/gasgiant/Camera-Shake/assets/52377879/8b79a579-3993-4713-a9f6-a3d20a4b88c3)

**Comment by gasgiant** [comment]
This whole package allocates on every shake. Wouldn't write it like this today, but this is how it is.

**Comment by StinkySteak** [comment]
maybe I will fork and create a pool for the default presets, what do you think? will it be difficult from your perspctive?

**Comment by gasgiant** [comment]
I don't think it would be too difficult to add pools to the whole system, not just for the default presets. You could create a number of pools indexed by the type of the shake. The system would return the shake instances to said pools when removing them from the active list, and the code creating shakes (in default presets or other scripts) would take instances from the pools, instead of creating new ones. One gotcha is you would probably have to look at the shake classes provided and add some form of reset logic, so they are safe to reuse.

**Comment by gasgiant** [comment]
I should also note that there might be more allocations inside the shake classes themselves. At the time I didn't care much about the allocations and used them liberally. I wasn't totally wrong - in most situations allocations in this package won't matter for performance. But I understand the desire to see zero in that GC Alloc column :-)

**Comment by StinkySteak** [comment]
This is now solved with my pull request
